### PR TITLE
Correct mutating op fallbacks

### PR DESCRIPTION
**Pull request by lgoettgens**
`Base.inv` and `AbstractAlgebra.inv` differ in handling Integers. The base version returns floats, while the AA version errors if it doesn't get a unit.

Similarly, `Base.div` and `AbstractAlgebra.div` differ. For this function, we need to either change the fallback of `div!` from `AbstractAlgebra.div` to `Base.div`, or change the docstring in a similar way as for `inv!`. Which of the two do we want here?

(In the any case, https://github.com/Nemocas/AbstractAlgebra.jl/pull/1816 needs to be adjusted to be consistent.)

### CODE REVIEW

**Comment by thofma** [comment]
Complicated. They need to fall back to the `Base` version, since this is what one should overload. But they should not fallback to the `Base` version for e.g. `BigInt`.

**Comment by lgoettgens** [comment]
> I am not sure a docstring inside AbstractAlgebra needs to use the AbstractAlgebra prefix.

Including this docstring into Oscar should prefix. I'll extend this a bit and maybe explain the difference there

File: src/fundamental_interface.jl [hunk 1]
```diff
@@ -328,7 +328,7 @@ neg!(a) = neg!(a, a)
     inv!(z, a)
     inv!(a)
 
-Return `inv(a)`, possibly modifying the object `z` in the process.
+Return `AbstractAlgebra.inv(a)`, possibly modifying the object `z` in the process.
```
[the remark this answers is not in the document]
Comment (fingolfin) [reply src/fundamental_interface.jl:331]: Ah I see you wrote that as a question. So yeah, I think `div!` should fallback to `AA.div!` (not `Base.div!`) and then its docstring should also be adjusted.

**Comment by lgoettgens** [comment]
I now adapted the docstrings of `inv!` and `div!`.
Furthermore, I encountered some methods that are attached to the wrong function. This shouldn't change any behavior due to the delegation from `AA.*` to `Base.*`, but makes it more consistent

File: src/Poly.jl [hunk 2]
```diff
@@ -1479,11 +1479,6 @@ function Base.divrem(f::PolyRingElem{T}, g::PolyRingElem{T}) where T <: RingElem
    return q, f
 end
 
-function Base.div(f::PolyRingElem{T}, g::PolyRingElem{T}) where T <: RingElement
-   q, r = divrem(f, g)
-   return q
-end
```
Comment (lgoettgens) [inline src/Poly.jl:1485]: redundant due to
https://github.com/Nemocas/AbstractAlgebra.jl/blob/c70d3b6900e4352ccffc22ff14b1f7b18516d21f/src/algorithms/GenericFunctions.jl#L77-L79

**Review by thofma — APPROVED** [review]

**Pull request merged**